poetry/packages/__init__.py

```diff
@@ -70,7 +70,7 @@ def dependency_from_pep_508(name):
             link = Link(path_to_url(os.path.normpath(os.path.abspath(link.path))))
         # wheel file
         if link.is_wheel:
-            m = re.match(r"^(?P<namever>(?P<name>.+?)-(?P<ver>\d.*?))", link.filename)
+            m = re.match(r"^(?P<namever>(?P<name>.+?)-(?P<ver>\d[^-]*))", link.filename)
```

Comment:
I think we can reuse the `wheel_file_re` from `poetry.utils.patterns`. That way we keep the consistency in the way we read wheel file names

Reply:
aha, thanks. I've updated to use `wheel_file_re`.